apps/aechannel/src/aesc_fsm.erl

```diff
@@ -866,14 +884,30 @@ channel_closed(cast, {?CHANNEL_CHANGED, _Info} = Msg, D) ->
 channel_closed(Type, Msg, D) ->
     handle_common_event(Type, Msg, discard, D).
 
+mutual_closed(enter, _, D) ->
+    keep_state(enter_closing(D));
+mutual_closed(cast, {?MIN_DEPTH_ACHIEVED, ChainId, ?WATCH_CLOSED, TxHash},
+              #data{ on_chain_id = ChainId
+                   , op = #op_min_depth{ tag = ?WATCH_CLOSED
+                                       , tx_hash = TxHash  % same tx hash
+                                         }} = D) ->
+    close(close_confirmed, D);
+mutual_closed(cast, {?SHUTDOWN_ERR, Msg}, D) ->
```

Comment:
I think this opens an attack vector (Bob is malicious):
1. Alice and Bob co-sign a mutual close, post it on-chain and enter `mutual_closed` state
2. Bob being malicious sends a `?SHUTDOWN_ERR` message to Alice and she rolls back to `open` state
3. Alice and Bob continue making off-chain updates. Bob spends some tokens and now Bob has less tokens than he did in step 1
4. Since the on-chain transaction had been posted on-chain - it gets included with the balances posted in step 1

Reply:
This is fixed now